hooks/_common.sh

```diff
@@ -217,6 +217,25 @@ function common::per_dir_hook {
     ((index += 1))
   done
 
+  # Lookup hook-config for modifiers that impact common behavior
+  local SKIP_CHDIR=false
+  IFS=";" read -r -a configs <<< "${HOOK_CONFIG[*]}"
+  for c in "${configs[@]}"; do
+    IFS="=" read -r -a config <<< "$c"
+    key=${config[0]}
+    value=${config[1]}
+
+    case $key in
+      --skip-chdir)
+        # this flag will skip pushing and popping directories
+        # delegating the responsibility to the hooked plugin/binary
+        if [ "$value" != "false" ]; then
+          SKIP_CHDIR=true
+        fi
```

Comment:
I'd prefer to compare with truthy value to avoid unexpectedness when user e.g. sets `no` instead of `false` by a mistake. Something like this (using modern `[[` allows to drop double quotes):
```suggestion
        if [[ ! $value || $value == true ]]; then
          SKIP_CHDIR=true
```